src/main/java/seedu/address/model/person/Person.java

```diff
@@ -93,6 +94,23 @@ public boolean isSamePerson(Person otherPerson) {
                 && otherPerson.getName().equals(getName());
     }
 
+    /**
+     * Returns an immutable group set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Group> getGroups() {
+        return Collections.unmodifiableSet(groups);
+    }
```

Comment:
I think we will have to make a few changes to this because when we eventually make the ui for this, we need to get a observable lsit version of this